File: backend/internal/service/assistant_types.go
package service

import "time"

const (
	AssistantSessionStatusActive = "active"

	AssistantMessageRoleUser      = "user"
	AssistantMessageRoleAssistant = "assistant"

	AssistantMessageKindChat                = "chat"
	AssistantMessageKindPlan                = "plan"
	AssistantMessageKindConfirmationRequest = "confirmation_request"
	AssistantMessageKindExecutionResult     = "execution_result"

	AssistantActionTypeDeployRef = "deploy_ref"

	AssistantIntentDeployRef        = "deploy_ref"
	AssistantIntentDeploymentStatus = "deployment_status"
	AssistantIntentQueryLogs        = "query_logs"
	AssistantIntentExplainIncident  = "explain_incident"
	AssistantIntentQueryTopology    = "query_topology"
	AssistantIntentReviewSystem     = "review_system"
	AssistantIntentQueryMetrics     = "query_metrics"
	AssistantIntentActivityTable    = "activity_table"
	AssistantIntentUnknown          = "unknown"

	AssistantPlanStatusDraft                = "draft"
	AssistantPlanStatusAwaitingConfirmation = "awaiting_confirmation"
	AssistantPlanStatusApproved             = "approved"
	AssistantPlanStatusExecuting            = "executing"
	AssistantPlanStatusCompleted            = "completed"
	AssistantPlanStatusFailed               = "failed"
	AssistantPlanStatusCancelled            = "cancelled"

	AssistantUIStateChat                 = "chat"
	AssistantUIStatePlanning             = "planning"
	AssistantUIStateAwaitingConfirmation = "awaiting_confirmation"
	AssistantUIStateExecuting            = "executing"
	AssistantUIStateCompleted            = "completed"
	AssistantUIStateFailed               = "failed"
)

type AssistantPlannerInput struct {
	UserID    string
	Role      string
	ProjectID string
	Content   string
	History   []AssistantMessageRecord
}

type AssistantPlannedIntent struct {
	Intent            string   `json:"intent"`
	Confidence        float64  `json:"confidence"`
	ProjectID         string   `json:"project_id,omitempty"`
	SourceRef         string   `json:"source_ref,omitempty"`
	RepoFullName      string   `json:"repo_full_name,omitempty"`
	TargetEnvironment string   `json:"target_environment,omitempty"`
	BindingHint       string   `json:"binding_hint,omitempty"`
	ServiceName       string   `json:"service_name,omitempty"`
	Window            string   `json:"window,omitempty"`
	Limit             int      `json:"limit,omitempty"`
	RequiresMutation  bool     `json:"requires_mutation"`
	MissingInputs     []string `json:"missing_inputs,omitempty"`
	Reason            string   `json:"reason,omitempty"`
}

type AssistantSessionRecord struct {
	ID            string    `json:"id"`
	ProjectID     *string   `json:"project_id,omitempty"`
	Title         string    `json:"title"`
	Status        string    `json:"status"`
	LastMessageAt time.Time `json:"last_message_at"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

type AssistantMessageRecord struct {
	ID          string         `json:"id"`
	SessionID   string         `json:"session_id"`
	Role        string         `json:"role"`
	Kind        string         `json:"kind"`
	Content     string         `json:"content"`
	ContentData map[string]any `json:"content_data,omitempty"`
	CreatedAt   time.Time      `json:"created_at"`
}

type AssistantMissingInput struct {
	Field   string `json:"field"`
	Prompt  string `json:"prompt"`
	Example string `json:"example,omitempty"`
}

type AssistantPlanRecord struct {
	ID                   string                  `json:"id"`
	ActionType           string                  `json:"action_type"`
	Status               string                  `json:"status"`
	Summary              string                  `json:"summary"`
	RiskLevel            string                  `json:"risk_level"`
	RequiresConfirmation bool                    `json:"requires_confirmation"`
	MissingInputs        []AssistantMissingInput `json:"missing_inputs,omitempty"`
	Plan                 map[string]any          `json:"plan"`
	ExpiresAt            *time.Time              `json:"expires_at,omitempty"`
	CreatedAt            time.Time               `json:"created_at"`
	UpdatedAt            time.Time               `json:"updated_at"`
}

type AssistantExecutionRecord struct {
	Status        string `json:"status"`
	DeploymentID  string `json:"deployment_id,omitempty"`
	RevisionID    string `json:"revision_id,omitempty"`
	BuildJobID    string `json:"build_job_id,omitempty"`
	CorrelationID string `json:"correlation_id,omitempty"`
	AgentID       string `json:"agent_id,omitempty"`
	Reason        string `json:"reason,omitempty"`
}

type AssistantConversationRecord struct {
	Session     AssistantSessionRecord    `json:"session"`
	Messages    []AssistantMessageRecord  `json:"messages"`
	PendingPlan *AssistantPlanRecord      `json:"pending_plan,omitempty"`
	UIState     string                    `json:"ui_state"`
	Execution   *AssistantExecutionRecord `json:"execution_result,omitempty"`
}

type CreateAssistantSessionCommand struct {
	UserID    string
	ProjectID string
	Title     string
}

type AssistantMessageCommand struct {
	UserID    string
	Role      string
	SessionID string
	ProjectID string
	Content   string
}

type ConfirmAssistantPlanCommand struct {
	UserID string
	Role   string
	PlanID string
}

type CancelAssistantPlanCommand struct {
	UserID string
	Role   string
	PlanID string
}

type DeployIntentPlan struct {
	ProjectID            string   `json:"project_id"`
	RepoFullName         string   `json:"repo_full_name,omitempty"`
	SourceRef            string   `json:"source_ref"`
	TargetEnvironment    string   `json:"target_environment"`
	BindingHint          string   `json:"binding_hint,omitempty"`
	DeploymentBindingID  string   `json:"deployment_binding_id,omitempty"`
	InternalServices     []string `json:"internal_services,omitempty"`
	ExecutionStrategy    string   `json:"execution_strategy"`
	MissingInputs        []string `json:"missing_inputs,omitempty"`
	Summary              string   `json:"summary"`
	RequiresConfirmation bool     `json:"requires_confirmation"`
	RiskLevel            string   `json:"risk_level"`
}

File: frontend/src/modules/assistant/assistant-types.ts
export type AssistantSession = {
  id: string;
  project_id?: string | null;
  title: string;
  status: string;
  last_message_at: string;
  created_at: string;
  updated_at: string;
};

export type AssistantMessage = {
  id: string;
  session_id: string;
  role: 'user' | 'assistant';
  kind: 'chat' | 'plan' | 'confirmation_request' | 'execution_result';
  content: string;
  content_data?: Record<string, unknown>;
  created_at: string;
};

export type AssistantHistoricalMatch = {
  id: string;
  project_id?: string;
  service_name: string;
  severity: string;
  title: string;
  body: string;
  correlation_id?: string;
  revision_id?: string;
  last_seen_at: string;
};

export type AssistantMissingInput = {
  field: string;
  prompt: string;
  example?: string;
};

export type AssistantActionPlan = {
  id: string;
  action_type: string;
  status: string;
  summary: string;
  risk_level: string;
  requires_confirmation: boolean;
  missing_inputs?: AssistantMissingInput[];
  plan: Record<string, unknown>;
  expires_at?: string | null;
  created_at: string;
  updated_at: string;
};

export type AssistantExecutionResult = {
  status: string;
  deployment_id?: string;
  revision_id?: string;
  build_job_id?: string;
  correlation_id?: string;
  agent_id?: string;
  reason?: string;
};

export type AssistantConversation = {
  session: AssistantSession;
  messages: AssistantMessage[];
  pending_plan?: AssistantActionPlan | null;
  ui_state: 'chat' | 'planning' | 'awaiting_confirmation' | 'executing' | 'completed' | 'failed';
  execution_result?: AssistantExecutionResult | null;
};

export type AssistantSessionListResponse = {
  items: AssistantSession[];
};

export type CreateAssistantSessionRequest = {
  project_id?: string;
  title?: string;
};

export type PostAssistantMessageRequest = {
  project_id?: string;
  content: string;
};
